src/stats/stats.jl

```diff
@@ -0,0 +1,158 @@
+export ExecutionStats, statshead, statsline
+
+const STATUS = Dict(:unknown => "unknown",
+                    :first_order => "first-order stationary",
+                    :max_eval => "maximum number of function evaluations",
+                    :max_time => "maximum elapsed time",
+                    :max_iter => "maximum iteration",
+                    :neg_pred => "negative predicted reduction",
+                    :unbounded => "objective function may be unbounded from below",
+                    :exception => "unhandled exception"
+                   )
+
+type ExecutionStats
+  status :: Symbol
+  solved :: Bool
+  tired :: Bool
+  stalled :: Bool
+  solution :: Vector # x
+  dual_vector :: Vector # λ
+  obj :: Float64 # f(x)
+  dual_feas :: Float64 # ‖∇f(x)‖ for unc, ‖P[x - ∇f(x)] - x‖ for bnd, etc.
+  primal_feas :: Float64 # ‖c(x)‖
```

Comment:
In what norm? What about inequality constraints?

Reply:
For now, I would recommend sticking with unconstrained problems.

Reply:
Clarified that it's the 2 norm.